File: Java-version/LeetCode/src/com/jooc/GazeOffer/Solution_11.java
package com.jooc.GazeOffer;

public class Solution_11 {
    public int minArray(int[] numbers) {
        int low = 0, high = numbers.length - 1;
        while(low < high){
            int mid = low + (high - low) / 2;
            if(numbers[mid] < numbers[high]){
                high = mid;
            }
            else if(numbers[mid] > numbers[high]){
                low = mid + 1;
            }
            else {
                high--;
            }
        }
        return numbers[low];
    }
}

File: Swift-version/LeetCode/LeetCode/Solutions/GazeOffer/Solution_11.swift
//
//  Solution_11.swift
//  LeetCode
//
//  Created by 齐旭晨 on 2021/3/26.
//

import Foundation

class Solution_11: Solution {
    func main() {
        print(minArray([3,0,1,1,1,1,1]))
    }
    
    func minArray(_ numbers: [Int]) -> Int {
        var left = 0, right = numbers.count-1
        while(left < right){
            let mid = (left+right)/2
            if numbers[mid] > numbers[right]{
                left = mid + 1
            }else if numbers[mid] < numbers[right]{
                right = mid
            }
            else{
                right -= 1
            }
        }
        return numbers[left]
    }
}
